src/app/app.vue

```diff
@@ -128,6 +129,36 @@ watch(
   },
 )
 
+const handleTrackingPermissionResult = (event: Event) => {
+  const customEvent = event as CustomEvent<string>
+  if (customEvent.detail === 'authorized') {
+    cookieControl.cookiesEnabledIds.value = [GTAG_COOKIE_ID]
```

Comment:
I think we can dynamically load the full list of optional cookies, something like:

```ts
const { moduleOptions } = useCookieControl()
moduleOptions.cookies.optional.map((x) => id.id)
```

It would be best to have [this utility function](https://github.com/dargmuesli/nuxt-cookie-control/blob/e71ef30bfed519272f00c217e5e6a4bc56f36d73/src/runtime/components/CookieControl.vue#L319) exported, but I don't find time to work on the cookie module currently, will maybe do this soon.